packages/loopring_v3/contracts/amm/libamm/AmmBlockReceiver.sol

```diff
@@ -74,11 +81,35 @@ library AmmBlockReceiver
         }
 
         // The closing AMM updates
-        S.processAmmUpdates(ctx, false);
+        S.approveAmmUpdates(ctx, false);
+
+        // Save the pool tokens to burn
+        uint totalSupply = S.totalSupply;
+
+        if (ctx.totalSupply > totalSupply) {
+            S.mint(address(this), ctx.totalSupply - totalSupply);
+        } else {
+            S.poolTokenToBurn = totalSupply - ctx.totalSupply;
+        }
```

Comment:
Have to do the pool token distribution after this I guess, though maybe have to watch out for joins/exits from the same user.